src/database/models/team_item.rs

```diff
@@ -100,11 +108,230 @@ impl TeamMember {
+                user_id,
+                name: m.member_name,
+                role: m.role,
+                permissions: Permissions::from_bits_truncate(m.permissions as u64),
+                accepted: m.accepted,
+            }))
+        } else {
+            Ok(None)
+        }
+    }
+
+    pub async fn insert(
+        &self,
+        transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+    ) -> Result<(), sqlx::error::Error> {
+        sqlx::query!(
+            "
+            INSERT INTO team_members (
+                id, user_id, member_name, role, permissions, accepted
+            )
+            VALUES (
+                $1, $2, $3, $4, $5,
+                $6
+            )
+            ",
+            self.id as TeamMemberId,
+            self.user_id as UserId,
+            self.name,
+            self.role,
+            self.permissions.bits() as i64,
+            self.accepted
+        )
+        .execute(&mut *transaction)
+        .await?;
+
+        Ok(())
+    }
+
+    pub async fn delete<'a, 'b, E>(
+        id: TeamId,
+        user_id: UserId,
+        executor: E,
+    ) -> Result<(), super::DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        sqlx::query!(
+            "
+            DELETE FROM team_members
+            WHERE (team_id = $1 AND user_id = $2 AND NOT role = $3)
+            ",
+            id as TeamId,
+            user_id as UserId,
+            crate::models::teams::OWNER_ROLE
+        )
+        .execute(executor)
+        .await?;
+
+        Ok(())
+    }
+
+    pub async fn edit_team_member<'a, 'b, E>(
+        id: TeamId,
+        user_id: UserId,
+        new_permissions: Option<i64>,
+        new_role: Option<String>,
+        new_accepted: Option<bool>,
+        executor: E,
+    ) -> Result<(), super::DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres> + Copy,
+    {
+        if let Some(permissions) = new_permissions {
+            sqlx::query!(
+                "
+                UPDATE team_members
+                SET permissions = $1
+                WHERE (team_id = $2 AND user_id = $3 AND NOT role = $4)
+                ",
+                permissions,
+                id as TeamId,
+                user_id as UserId,
+                crate::models::teams::OWNER_ROLE.to_string(),
+            )
+            .execute(executor)
+            .await?;
+        }
+
+        if let Some(role) = new_role {
+            sqlx::query!(
+                "
+                UPDATE team_members
+                SET role = $1
+                WHERE (team_id = $2 AND user_id = $3 AND NOT role = $4)
+                ",
+                role,
+                id as TeamId,
+                user_id as UserId,
+                crate::models::teams::OWNER_ROLE.to_string(),
+            )
+            .execute(executor)
+            .await?;
+        }
+
+        if let Some(accepted) = new_accepted {
```

Comment:
This should only ever work for accepting a request.  You should not be able to change accepted to false; leaving a team should be deleting the team member.